### PR TITLE
[MIG][11.0] - stock picking mass assign

### PR DESCRIPTION
**Pull request by ougc27**
Hello @pedrobaeza 

This PR migrates the module stock_picking_mass_assign to version 11. Please review it.

Thanks & Regards!

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Sorry, but I'm not a user of this module right now. I only handle the migration issues. Wait for other reviewers or review others PRs and ask in exchange that they review yours.

**Comment by fanha99** [comment]
hi, I tested on the runbot.
- I try to mass check availability of some Delivery Orders which is enough product
- then try to unreserved some of them.
- mass check availability to them (more than 1)
- (tree view) there is some Delivery Orders state is still waiting,
- (form view) Delivery Orders, there is only validate button, no **check availability** button, and no **unreserved button**, the move line is reserved zero.
- click on move line, the move line form popup, the product is now reserved.
b1
b2

**Comment by ougc27** [comment]
Hi @fanha99 

Thank you for your review

**Comment by fanha99** [comment]
Hi @ougc27,
Can you take a look at https://github.com/odoo/odoo/pull/21939. I think it is relative with the bug I found.

**Review by alan196 — APPROVED** [review]
Tested functionally and works perfect 👍

**Review by legalsylvain — CHANGES_REQUESTED** [review]
Hi @ougc27, thanks for porting this module. I guess you missed some commits during the port of this module. This module has more options in v8, and has been renamed into ``stock_picking_mass_action``. 
could you cherry pick this changes ? kind regards.
https://github.com/OCA/stock-logistics-workflow/tree/8.0/stock_picking_mass_action

**Comment by pedrobaeza** [comment]
We have migrated it to 10.0 in #407

File: stock_picking_mass_assign/tests/__init__.py [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# -*- coding: utf-8 -*-
```
Comment (hieulucky111) [inline stock_picking_mass_assign/tests/__init__.py:1]: remove it.

File: stock_picking_mass_assign/tests/test_check_assign_all.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+                'procure_method': 'make_to_stock',
+            })],
+        })
+        picking.action_confirm()
+        return picking
+
+    def test_10_check_assign_partially_available(self):
+        pickings = self.create_stock_picking()
+        pickings |= self.create_stock_picking()
+        pickings |= self.create_stock_picking()
```
Comment (hieulucky111) [inline stock_picking_mass_assign/tests/test_check_assign_all.py:47]: You can try to use a common function:
```python
def _generate_pickings(self, picking_number):
    pickings = self.picking_obj
    for index in range(picking_number):
        pickings |= self.create_stock_picking()
    return pickings
```

File: stock_picking_mass_assign/tests/test_check_assign_all.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+            active_ids=pickings.ids, process_picking=True).create({}).check()
+        self.assertEquals(
+            pickings.mapped('state'),
+            ['done', 'done', 'assigned'])
+        return True
+
+    def test_20_check_assign_no_orders(self):
+        pickings = self.create_stock_picking()
+        pickings |= self.create_stock_picking()
+        pickings |= self.create_stock_picking()
```
Comment (hieulucky111) [inline stock_picking_mass_assign/tests/test_check_assign_all.py:58]: Same as above. Consider using function.

File: stock_picking_mass_assign/tests/test_check_assign_all.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        with self.assertRaisesRegexp(
+                ValidationError, _('No selected delivery orders')):
+            self.check_assign_all_wiz.with_context().create({}).check()
+        return True
+
+    def test_30_check_all(self):
+        pickings = self.create_stock_picking()
+        pickings |= self.create_stock_picking()
+        pickings |= self.create_stock_picking()
```
Comment (hieulucky111) [inline stock_picking_mass_assign/tests/test_check_assign_all.py:67]: Same above.

File: stock_picking_mass_assign/wizard/check_assign_all.py [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+class CheckAssignAll(models.TransientModel):
+    _name = 'stock.picking.check.assign.all'
+    _description = 'Delivery Orders Check Availability'
+
+    check_availability = fields.Boolean(
+        'Check Availability',
+        default=True,
+        help="""check this box if you want to check the availability of """
+        """the selected Delivery Orders.""")
```
Comment (hieulucky111) [inline stock_picking_mass_assign/wizard/check_assign_all.py:18]: I think you should be use double quote instead triple quote as `"check this box if you want to check the availability of the selected Delivery Orders."`

**Review by hieulucky111 — CHANGES_REQUESTED** [review]
Please check my review.

**Comment by liebana** [comment]
Hello:

https://github.com/OCA/stock-logistics-workflow/tree/10.0/stock_picking_mass_action and https://github.com/OCA/stock-logistics-workflow/tree/10.0/stock_picking_mass_assign are duplicated in v10, same functionality. It seems that stock_picking_mass_action, migrated by Tecnativa as referred by @pedrobaeza, is more updated and we're interested in the migration to v11. We should open a new PR (this one is referred to stock_picking_mass_assign), right?

**Comment by pedrobaeza** [comment]
Yes

**Comment by CasVissers-360ERP** [comment]
LGTM

**Comment by legalsylvain** [comment]
Indeed stock_picking_mass_action is the most complete module. This one should be migrated.
so, closing.

**Pull request closed without merging**